### PR TITLE
rust-analyzer: 2021-07-12 -> 2021-07-26

### LINKED ISSUE #131329: rust-analyzer: 2021-07-12 -> 2021-07-19
###### Checklist

- [x] Checked the [nixpkgs master branch](https://github.com/NixOS/nixpkgs)

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

###### Project name
`nix search` name: rust-analyzer

current version: 2021-07-12
desired version: 2021-07-19

###### Notify maintainers

maintainers: @oxalica 

###### Note for maintainers

Please tag this issue in your PR.

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Update to latest upstream release 2021-07-19

Change log: 
- https://github.com/rust-analyzer/rust-analyzer/releases/tag/2021-07-26
- https://github.com/rust-analyzer/rust-analyzer/releases/tag/2021-07-19

Fixes #131329

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Mic92** [comment]
Result of `nixpkgs-review pr 131502` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>vscode-extensions.matklad.rust-analyzer</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>rust-analyzer</li>
    <li>rust-analyzer-unwrapped</li>
  </ul>
</details>

**Comment by Mic92** [comment]
@GrahamcOfBorg build vscode-extensions.matklad.rust-analyzer

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 131502` at 811238cf run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/aa5e9465cc2aabb19494cb33d9b28ff8">vscode-extensions.matklad.rust-analyzer</a></li>
  </ul>
</details>
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>rust-analyzer</li>
    <li>rust-analyzer-unwrapped</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

**Comment by oxalica** [comment]
2021-07-26 is released new.

**Review by oxalica — CHANGES_REQUESTED** [review]
Seems `package.json` is updated. Node-packages should be regenerated by
`cd pkgs/development/node-packages; ./generate.sh`
Or the vscode extension will fail to build.

(If you use updater script `pkgs/development/tools/rust/rust-analyzer/update.sh`. It will automatically handle it.)

**Comment by fabaff** [comment]
Äh, I didn't include the updated file. Thanks for the feedback.

**Comment by oxalica** [comment]
The vscode extension still doesn't build.
Log: https://gist.github.com/oxalica/42c2ab9431dd592fe9f5181c80b16a19

**Comment by Mic92** [comment]
> The vscode extension still doesn't build.
> Log: https://gist.github.com/oxalica/42c2ab9431dd592fe9f5181c80b16a19

Did it build before this update?

**Comment by oxalica** [comment]
>>   The vscode extension still doesn't build.
>>    Log: https://gist.github.com/oxalica/42c2ab9431dd592fe9f5181c80b16a19
>
>Did it build before this update?

Yes. I'm on 67efb9a7634a38e9b9de27064146bb1aee9770af. The error message seems weird. I'm not sure what's happening.

Here is the derivation path.
```
repo/fork/nixpkgs ((67efb9a7634...)) > nix build -f . vscode-extensions.matklad.rust-analyzer -vL
building '/nix/store/sv26adzmwmw3zsr7mvhw6g0sm6lvj682-rust-analyzer-0.4.0-dev.drv'...
```

File: pkgs/misc/vscode-extensions/rust-analyzer/build-deps/package.json [hunk 1]
```diff
@@ -6,20 +6,20 @@
...
-    "mocha": "^8.2.1",
-    "rollup": "^2.34.2",
-    "tslib": "^2.0.3",
-    "typescript": "^4.1.2",
+    "mocha": "^9.0.2",
+    "rollup": "^2.51.1",
+    "tslib": "^2.3.0",
+    "typescript": "^4.3.5",
     "typescript-formatter": "^7.2.2",
     "vsce": "=1.88.0",
```
Comment (yaymukund) [inline pkgs/misc/vscode-extensions/rust-analyzer/build-deps/package.json:24]: I do not know if it matters, but this is [different from the version specified upstream](https://github.com/rust-analyzer/rust-analyzer/blob/2021-07-26/editors/code/package.json#L59). How is this file generated / what is it meant to track?

**Comment by yaymukund** [comment]
I briefly tried to debug this. It appears that vsce is an optional dependency that gets skipped?

> `[K[?25h[37;40mnpm[0m [0m[30;43mWARN[0m [0m[35moptional[0m SKIPPING OPTIONAL DEPENDENCY: vsce@^1.95.1 (node_modules/vsce):`

That output sure is strange though.

File: pkgs/misc/vscode-extensions/rust-analyzer/build-deps/package.json (hunk 1 above)
Comment (fabaff) [reply pkgs/misc/vscode-extensions/rust-analyzer/build-deps/package.json:24]: It's generated by the update script.

**Comment by Mic92** [comment]
out dated.

**Pull request closed without merging**